src/ImageSharp/PixelFormats/PixelBlenders/PorterDuffFunctions.cs

```diff
@@ -243,12 +288,9 @@ public static Vector4 Over(Vector4 destination, Vector4 source, Vector4 blend)
     [MethodImpl(MethodImplOptions.AggressiveInlining)]
     public static Vector256<float> Over(Vector256<float> destination, Vector256<float> source, Vector256<float> blend)
     {
-        const int blendAlphaControl = 0b_10_00_10_00;
-        const int shuffleAlphaControl = 0b_11_11_11_11;
-
         // calculate weights
-        Vector256<float> sW = Avx.Shuffle(source, source, shuffleAlphaControl);
-        Vector256<float> dW = Avx.Shuffle(destination, destination, shuffleAlphaControl);
+        Vector256<float> sW = Avx.Shuffle(source, source, ShuffleAlphaControl);
```

Comment:
Since it's the same input twice, using `Avx.Permute` should be preferred. The same shuffle control is used.

Reply:
`vpermilps` is 1 byte larger than `vshufps`.  The only reason to prefer it is that it supports a memory operand where shuffle doesn't, but that doesn't apply here.

Reply:
It's 1-byte larger but also preferred by Clang/GCC in practically all cases (both Intel and AMD have their instruction selection metadata set to prefer it).

There have been CPUs in the past (Knight's Landing most notably) where there was actual cost difference between them. Any CPU that decides to emulate YMM based operations via 2x 128-bit ops internally may likewise see worse performance.

It ultimately shouldn't make too much a difference, but I'd typically side with what Intel/AMD have made Clang/GCC emit.

Reply:
> (both Intel and AMD have their instruction selection metadata set to prefer it).

I don't know what this means.  The two instructions mostly do different things, with one overlapping case.  Can you point me at the metadata in question?  And any example where Clang does the substitution?

> There have been CPUs in the past (Knight's Landing most notably) where there was actual cost difference between them. Any CPU that decides to emulate YMM based operations via 2x 128-bit ops internally may likewise see worse performance for shuffle.

Ignoring Knight's Landing since it's not a real CPU, this argument doesn't make any sense to me.  This is an in-lane operation, so there's no reason splitting it in two would affect one and not the other.  In fact, the two existing architectures that do split AVX into 2x128 (Zen+ and Gracemont) have identical measured latency and throughput numbers.

Not that 1 byte here or there is especially worth arguing over, but since you've suggested it "should be preferred", I'd like to see something concrete.

Reply:
You can trivially see this in godbolt via:
```cpp
#include <immintrin.h>

__m256 Permute(__m256 x)
{
    return _mm256_shuffle_ps(x, x, 0xFF);
}
```

and that Clang and GCC both universally replace this with `vpermilps` regardless of ISAs or target `-march=...`.

I'll see if I can dig up, again, exactly where the instruction selection for this happens, but might take me a bit given the size/complexity of LLVM in general.

Reply:
Well that's interesting.  At least trusty old MSVC does what it's told 😉

https://godbolt.org/z/3eTKGqchf

I'd appreciate it if you can find the reference for why they do it.  I've seen Clang make some odd decisions before, but with GCC doing it too there's got to be something.